Grbl_Esp32/Grbl_Esp32.ino

```diff
@@ -42,6 +42,22 @@ Spindle *spindle;
 
 
 void setup() {
+#ifdef USE_I2S_OUT
```

Comment:
I will push this as is, but in the future it would be nice to have this block of code as a subroutine so the details are not exposed in the top level file.

Reply:
I don't think it's beautiful either.
This sequence is positioned one layer above the i2s out layer.
Shall I put the little sequence in ino as a function?
It also didn't seem beautiful to have a dedicated function for using i2s out in ino.
Also, I didn't want to include an implementation that uses i2s out within i2s out.
After much deliberation, this is the current implementation for now.

Reply:
So far, neither define nor delay should be necessary.
So it's probably a good idea to check if it's OK to remove it, and if it's OK, clean it up.

Reply:
One way to finesse the problem is to overload i2s_out_init() with a version that has no parameters, whose body is the code below.  It would go in i2s_out.cpp.  If USE_I2S_OUT is not defined, a do-nothing version of the function would be supplied.

This is just one small step in the quest to rid the code of ifdefs all over the place.  A long time ago in a galaxy far away, compilers and linkers were not smart enough to remove dead code automatically, so you needed the preprocessor to help.  Not anymore.

Reply:
I see, Master. You are absolutely right.
I forgot that I'm writing in C++, I use the Force.